core/src/proof.rs

```diff
@@ -6,12 +6,30 @@ use crate::trie::{
 
 use bitvec::prelude::*;
 
+/// Wrapper for a terminal node, it will store the LeafData if it is a leaf node,
+/// and just the KeyPath to that terminal if it is a terminator node
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum PathProofTerminal {
+    Leaf(LeafData),
+    Terminator(KeyPath),
```

Comment:
It doesn't make sense for a Terminator to have a KeyPath. It can have a bit-path, but not a full key-path, unless it's at depth 256.